logstage/logstage-core/src/main/scala/izumi/logstage/api/Log.scala

```diff
@@ -157,7 +157,17 @@ object Log {
   final case class Message(template: StringContext, args: LogContext) {
     def argsMap: Map[String, Set[Any]] = args.map(kv => (kv.name, kv.value)).toMultimap
 
-    def ++(that: Message): Message = Message(StringContext(template.parts ++ that.template.parts: _*), args ++ that.args)
+    def ++(that: Message): Message = {
+      // this ain't pretty but avoids multiple iterations on the same parts collection compared to using .init/.tail
+      val (thisInit, thisTails) = template.parts.splitAt(template.parts.length - 1)
+      val thisTail = thisTails.head
+
+      val (thatHeads, thatTail) = that.template.parts.splitAt(1)
+      val thatHead = thatHeads.head
```

Comment:
@an-tex Could you please make it work if the right Message is empty (`Message(StringContext(), Nil)`) or consists of one part (`Message("")`), and add a test for that?